interview-questions.md

```diff
@@ -8,42 +8,76 @@ Then, research the question to expand on your answer. Even if you feel you have
+    
+    Encapsulation - Encapsulation is to contain everything within the object or a single unit. its used to protect data structures so they don't get used or access elsewhere. 
+    
+    Abstraction - the idea is to hide data. 
+        The act of representing the essential information and hiding the unncessary information what the intent at hand.
+
+    Inheritance - the idea is reusability.  The children classes/instances will inherit characteristics like data and methods from their parent class. 
+
 2. What is the difference between a Float and an Integer in Ruby?
 
-Your answer:
+Your answer: Float numbers have decimals and integers in ruby do not have decimals. 
 
-Researched answer:
+Researched answer:  
+    There are these classes of numbers in Ruby: Numeric Integer, Fixnum, Bignum, Float, Complex, Rational, BigDecimal (Standard Library).  Each one of these classes will carry and represent the numbers a specific way.  It is important to know which class of numbers to use in which time.. for example.. when using Bignum to get decimal numbers.. its sometimes slower to process than using float numbers.
 
 3. Ruby has an implicit return. What does that mean?
 
-Your answer:
+Your answer: 
+    Implicit return means that the code block automatically returns something without actually declaring a return.  
 
 Researched answer:
+    Every block in ruby will return the last line automatically so it is common that people not to use a return to get their code a more cleaner looks.  There are advantages and disadvantages to using implicit return or to use a regular return.  It is important to know what methods are used exactly so that we can avoid issues with the code moving forward when trying to use implicit returns. 
 
 4. What is a block in Ruby?
 
 Your answer:
+    A ruby block is the block of code that goes between do and end. Its almost like the body of the code. and the method that calls it is the head of the block of code. method calls signals what to do with the code block.
 
 Researched answer:
+    A block in ruby is any chuck of code between do and end. 
+    Just having a block doesn't do anything, there must be a method call that goes before do.
+    The method call is kind of like the head of the block, and the block is the body of the code.
+    There could be single line block or multiple lines block.
+    Instead of using "do" and "end", we could wrap the block inside curly braces.
+    Blocks could also contain parameters.
 
-1. How do you extract a value in a Ruby hash?
+5. How do you extract a value in a Ruby hash?
 
-Your answer:
+Your answer: 
+    hash.fetch(key), this is a built in method which returns the value of the key inside of a hash.
 
 Researched answer:
+    to make a function out of this.. it is important to input the hash and the key.
+    for example: def retrieve_value(hash, key).
+
+    I'm sorry.. I'm just so tired. hahaha. short answers only. 
```

Comment:
Short answers or not, you still did a really great job in this section.  When possible though, make sure to expand.  I really like the detail you add into some of these researched answers.